src/control/server/storage/scm/ipmctl.go

```diff
@@ -55,49 +56,17 @@ type (
 )
 
 func (sv semVer) String() string {
-	sb := new(strings.Builder)
-	for _, e := range sv {
-		sb.WriteString(e + ".")
-	}
-
-	return strings.TrimRight(sb.String(), ".")
+	return strings.Join(sv, ".")
 }
 
-func (sv semVer) validate(log logging.Logger) error {
+func validateIpmctlVer(log logging.Logger, sv semVer) error {
```

Comment:
This is better, but what I was suggesting is that this function should accept a `semVer` and a `[]semVer` and return a boolean indicating whether or not the semVer is in the supplied list. That makes it more flexible and easier to test. You could repurpose it later to check for known-good versions, for example.

I think it would also be good to do the logging outside of the function.

Reply:
to me it sounds like that should be a function attached to the semVer type that accepts a list and validates existence in that list. but happy to follow your preference

Reply:
what I described above is also in line with what we have implemented with ranks I think ..InList or somesuch